store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -2743,6 +2852,269 @@ public String getServiceName() {
 
     }
 
+    class MainBatchDispatchRequestService extends ServiceThread {
+
+        private final ExecutorService batchDispatchRequestExecutor;
+
+        public MainBatchDispatchRequestService() {
+            batchDispatchRequestExecutor = new ThreadPoolExecutor(
+                    DefaultMessageStore.this.getBrokerConfig().getBatchDispatchRequestThreadPoolNums(),
+                    DefaultMessageStore.this.getBrokerConfig().getBatchDispatchRequestThreadPoolNums(),
+                    1000 * 60,
+                    TimeUnit.MICROSECONDS,
+                    new LinkedBlockingQueue<>(4096),
+                    new ThreadFactoryImpl("BatchDispatchRequestServiceThread_"),
+                    new RejectedExecutionHandler() {
+                        @Override
+                        public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+                            try {
+                                LOGGER.warn("Task {} is blocking put into the workQueue", r);
+                                executor.getQueue().put(r);
+                            } catch (InterruptedException e) {
+                                LOGGER.error("Task {} failed to put into the workQueue", r);
+                            }
+                        }
+                    });
+        }
+
+        private void pollBatchDispatchRequest() {
+            if (!batchDispatchRequestQueue.isEmpty()) {
+                BatchDispatchRequest task = batchDispatchRequestQueue.poll();
+                batchDispatchRequestExecutor.execute(() -> {
+                    ByteBuffer tmpByteBuffer = task.byteBuffer.duplicate();
```

Comment:
batchDispatchRequestExecutor  execute  runnable  must be have try catch final protect

Reply:
Is that ok?

![image](https://user-images.githubusercontent.com/72536832/216903825-95ec619c-2578-4233-a221-284b3b0af483.png)

Reply:
Give you the demo, you can think about different

```
package org.apache.rocketmq.test;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

import org.apache.rocketmq.common.ThreadFactoryImpl;

public class Test {

    
    public static void main(String[] args) {
        good(false);
        good(true);
        bad(false);
        bad(true);

    }

    public static void good(boolean throwException) {
        ExecutorService executor = new ThreadPoolExecutor(1, 1,
                1000 * 60,
                TimeUnit.MICROSECONDS,
                new LinkedBlockingQueue<>(2),
                new ThreadFactoryImpl("GoodBatchDispatchRequestServiceThread_"),
                new ThreadPoolExecutor.CallerRunsPolicy());
        for (int i = 0; i < 15; i++) {
            executor.execute(() -> {
                try {
                    TimeUnit.SECONDS.sleep(1);
                    System.out.println("ThreadName:" + Thread.currentThread().getName());
                    if (throwException)
                        throw new RuntimeException("test");
                } catch (Exception e) {

                }
            });
        }

        for (int i = 0; i < 5; i++) {
            executor.execute(() -> {
                System.out.println("testbb");
            });
        }
        System.out.println("test-cccccccc");

    }

    public static void bad(boolean throwException) {
        ExecutorService executor = new ThreadPoolExecutor(1, 1,
                1000 * 60,
                TimeUnit.MICROSECONDS,
                new LinkedBlockingQueue<>(2),
                new ThreadFactoryImpl("BadBatchDispatchRequestServiceThread_"));
        for (int i = 0; i < 15; i++) {
            executor.execute(() -> {
                try {
                    TimeUnit.SECONDS.sleep(1);
                } catch (InterruptedException e) {

                }
                System.out.println("ThreadName:" + Thread.currentThread().getName());
                if (throwException)
                    throw new RuntimeException("test");
            });
        }

        for (int i = 0; i < 5; i++) {
            executor.execute(() -> {
                System.out.println("testbb");
            });
        }
        System.out.println("test-cccccccc");

    }
}
```

Reply:
The differences I think are: 
1) good() has a deny reject CallerRunsPolicy, and the calling thread main is responsible for performing the overflowed task. Bad() does not have a reject policy, and overflowed tasks are discarded. 
2) Exceptions thrown within the scope of try will be catch. Exceptions thrown outside the scope of the try are thrown to a higher level and interrupt the current program, and seem to cause Executor to create a new thread (the previous thread may have been destroyed)

Reply:
you can refactor related code as soon as possible. @nowinkeyy